app/process/handler/consensus-message/consensus-message.go

```diff
@@ -1,3 +1,110 @@
 package consensusmessage
 
-// TODO: Consensus message event handler
+import (
+	"encoding/hex"
+	"errors"
+	"fmt"
+	"github.com/ethereum/go-ethereum/crypto"
+	"github.com/golang/protobuf/proto"
+	"github.com/limechain/hedera-eth-bridge-validator/app/domain/repositories"
+	"github.com/limechain/hedera-eth-bridge-validator/app/persistence/message"
+	"github.com/limechain/hedera-eth-bridge-validator/config"
+	validatorproto "github.com/limechain/hedera-eth-bridge-validator/proto"
+	log "github.com/sirupsen/logrus"
+	"strings"
+)
+
+type ConsensusMessageHandler struct {
+	repository     repositories.MessageRepository
+	validAddresses []string
+}
+
+func NewConsensusMessageHandler(r repositories.MessageRepository) *ConsensusMessageHandler {
+	return &ConsensusMessageHandler{
+		repository:     r,
+		validAddresses: config.LoadConfig().Hedera.Handler.ConsensusMessage.Addresses,
+	}
+}
+
+func (cmh ConsensusMessageHandler) Handle(payload []byte) {
+	err := cmh.handlePayload(payload)
+	if err != nil {
+		log.Errorf("Error - could not handle payload: [%s]", err)
+		return
+	}
+}
+
+func (cmh ConsensusMessageHandler) handlePayload(payload []byte) error {
+	m := &validatorproto.TopicSignatureMessage{}
+	err := proto.Unmarshal(payload, m)
+	if err != nil {
+		return errors.New(fmt.Sprintf("Failed to unmarshal topic signature message. - [%s]", err))
+	}
+
+	ctm := &validatorproto.CryptoTransferMessage{
+		TransactionId: m.TransactionId,
+		EthAddress:    m.EthAddress,
+		Amount:        m.Amount,
+		Fee:           m.Fee,
+	}
+
+	log.Printf("[%s] signature", m.GetSignature())
```

Comment:
If we are looking at the logs, we are going to see random strings of signatures.
I think that first of all we should not log the signatures, and second, if we logging anything we should add transaction ID to every log. Otherwise, we won't know what the context is. Think of how the logs will look like if there are 10 txns/sec.